packages/react-juce/src/components/Slider.tsx

```diff
@@ -0,0 +1,284 @@
+
+        const rotateAngle = ((180 + 180 * (1 - arcSize)) * Math.PI) / 180;
+
+        const deltaX = x + radius;
+        const deltaY = y + radius;
+
+        ctx.translate(deltaX, deltaY);
+        ctx.rotate(rotateAngle);
+        ctx.translate(-deltaX, -deltaY);
+
+        ctx.stroke();
+        ctx.resetTransform();
+    }
+}
+
+export interface SliderProps {
+    value: number
+    sensitivity?: number
+    onChange: (value: number) => void
+    onDraw: (
+        ctx: CanvasRenderingContext2D,
+        width: number,
+        height: number,
+        value: number
+    ) => void
+    mapDragGestureToValue: (
+        mouseDownX: number,
+        mouseDownY: number,
+        sensitivity: number,
+        valueAtDragStart: number,
+        dragEvent: SyntheticMouseEvent
+    ) => number
+}
+
+type SliderState = {
+    width: number,
+    height: number
+}
+
+//TODO: Remove use of any for props type once blueprint ViewProps is typed
+//TODO: Add simple bipolar (i.e. center split) gesture/draw helpers
+export class Slider extends Component<SliderProps | any, SliderState> {
+    static linearHorizontalGestureMap = _linearHorizontalGestureMap;
+    static linearVerticalGestureMap = _linearVerticalGestureMap;
+    static rotartyGestureMap = _rotaryGestureMap;
+
+    static drawLinearHorizontal = _drawLinearHorizontalSlider;
+    static drawLinearVertical = _drawLinearVerticalSlider;
+    static drawRotary = _drawRotarySlider;
+
+    private _valueAtDragStart = 0.0;
+    private _mouseDownX = 0;
+    private _mouseDownY = 0;
+
+    constructor(props: PropsWithChildren<SliderProps | any>) {
+        super(props);
+
+        this.state = {
+            width: 0,
+            height: 0
+        };
+    }
+
+    _onMeasure = (e: any) => {
+        this.setState({
+            width: e.width,
+            height: e.height
+        });
+    }
+
+    _onMouseDown = (e: SyntheticMouseEvent) => {
+        this._valueAtDragStart = this.props.value;
+        this._mouseDownX = e.x;
+        this._mouseDownY = e.y;
+
+        if (typeof this.props.onMouseDown === 'function') {
+            this.props.onMouseDown(e);
+        }
+    }
+
+    _onMouseDrag = (e: SyntheticMouseEvent) => {
+        if (typeof this.props.mapDragGestureToValue === 'function') {
+            const value = this.props.mapDragGestureToValue(
+                            this._mouseDownX,
+                            this._mouseDownY,
+                            this.props.sensitivity,
+                            this._valueAtDragStart,
+                            e
+                          );
+
+            if (typeof this.props.onChange === 'function') {
+                this.props.onChange(value);
+            }
+        }
+
+        if (typeof this.props.onMouseDrag === 'function') {
+            this.props.onMouseDrag(e);
+        }
+    }
+
+    _onDraw = (ctx: CanvasRenderingContext2D) => {
+        if (typeof this.props.onDraw === 'function') {
```

Comment:
No need to make our own `_onDraw` here if the only goal is to call the incoming `props.onDraw`. We can just unpack `props` into the `Canvas` below and if there's an onDraw there, it will get passed through.

That said, I think the slider should probably have some default `onDraw` in which case this would be the case to do it....

```
if (typeof this.props.onDraw === 'function') {
  return this.props.onDraw(....)
}

// We'll fall through to here if we don't have an on draw
_rotaryDrawHelper(ctx, this.state.width, this.state.height, this.props.value);
```

Reply:
Yeah OK cool so keep `_onDraw` but just for the purposes of a fallback.